tests/test_s3.py

```diff
@@ -1543,6 +1543,289 @@ def test_has_statement_policy(self):
+        try:
+            response = client.get_public_access_block(
+                Bucket=bname)['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                response = None
+            else:
+                raise
+        if response:
+            for key in response.keys():
+                self.assertEqual(response[key], True)
+
+    def test_set_public_block_disable_all(self):
+        bname = 'mypublicblock'
+
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3,
+            "S3_AUGMENT_TABLE",
+            [('get_public_access_block', 'PublicBlock',
+            None, None, 's3:GetBucketPublicAccessBlock')],
+        )
+        self.patch(
+            s3.S3.resource_type,
+            "enum_spec",
+            ('list_buckets', "Buckets[?Name=='{}']".format(bname), None)
+        )
+        session_factory = self.replay_flight_data("test_s3_public_block_disable_all")
+        session = session_factory()
+        client = session.client("s3")
+
+        p = self.load_policy(
+            {
+                "name": "CheckForPublicBlocks-Present",
+                "resource": "s3",
+                "filters": [
+                    {
+                        "type": "check-public-block",
+                        "scope": "Any",
+                        "state": "present"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "set-public-block",
+                        "scope": "All",
+                        "state": "disable"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+
+        # Test that there was a bucket with missing public blocks
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        # Verbose test to make sure that the public blocks are enabled on the buckets
+        try:
+            response = client.get_public_access_block(
+                Bucket=bname)['PublicAccessBlockConfiguration']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                self.assertEqual(response, None)
+            else:
+                raise
+        if response:
+            for key in response.keys():
+                self.assertEqual(response[key], False)
+
+    def test_set_public_block_enable_one(self):
+        bname = 'mypublicblock'
+
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3,
+            "S3_AUGMENT_TABLE",
+            [('get_public_access_block', 'PublicBlock',
+            None, None, 's3:GetBucketPublicAccessBlock')],
+        )
+        self.patch(
+            s3.S3.resource_type,
+            "enum_spec",
```

Comment:
this isn't actually helpful, its a client side trick, equivalent to adding a filter to your policy of Name: xyz

Reply:
Good call, hadn't thought of it like that. Makes sense to use the policies native filtering capabilities. Resolved in ed8064fe9ca962eca70348b6dc7587567e90ade1